code/modules/power/terminal.dm

```diff
@@ -42,3 +42,37 @@
 		qdel(src)
 		return
 	..()
+
+/obj/machinery/power/proc/make_terminal(mob/user)

```

Comment:
Make sure the terminal is cardinally adjacent to the thing it is being added to.